compiler/rustc_attr_parsing/src/validate_attr.rs

```diff
@@ -150,98 +149,17 @@ fn is_attr_template_compatible(template: &AttributeTemplate, meta: &ast::MetaIte
-
-            // If the `attr_item`'s span is not from a macro, then just suggest
-            // wrapping it in `unsafe(...)`. Otherwise, we suggest putting the
-            // `unsafe(`, `)` right after and right before the opening and closing
-            // square bracket respectively.
-            let diag_span = attr_item.span();
-
-            // Attributes can be safe in earlier editions, and become unsafe in later ones.
-            //
-            // Use the span of the attribute's name to determine the edition: the span of the
-            // attribute as a whole may be inaccurate if it was emitted by a macro.
-            //
-            // See https://github.com/rust-lang/rust/issues/142182.
-            let emit_error = match unsafe_since {
-                None => true,
-                Some(unsafe_since) => path_span.edition() >= unsafe_since,
-            };
-
-            if emit_error {
-                psess.dcx().emit_err(errors::UnsafeAttrOutsideUnsafe {
-                    span: path_span,
-                    suggestion: errors::UnsafeAttrOutsideUnsafeSuggestion {
-                        left: diag_span.shrink_to_lo(),
-                        right: diag_span.shrink_to_hi(),
-                    },
-                });
-            } else {
-                psess.buffer_lint(
-                    UNSAFE_ATTR_OUTSIDE_UNSAFE,
-                    path_span,
-                    id,
-                    BuiltinLintDiag::UnsafeAttrOutsideUnsafe {
-                        attribute_name_span: path_span,
-                        sugg_spans: (diag_span.shrink_to_lo(), diag_span.shrink_to_hi()),
-                    },
-                );
-            }
-        }
-
-        // - Normal builtin attribute
-        // - Writing `#[unsafe(..)]` is not permitted on normal builtin attributes
-        (None | Some(AttributeSafety::Normal), Safety::Unsafe(unsafe_span)) => {
-            psess.dcx().emit_err(errors::InvalidAttrUnsafe {
-                span: unsafe_span,
-                name: attr_item.path.clone(),
-            });
-        }
-
-        // - Normal builtin attribute
-        // - No explicit `#[unsafe(..)]` written.
-        (None | Some(AttributeSafety::Normal), Safety::Default) => {
-            // OK
-        }
-
-        (
-            Some(AttributeSafety::Unsafe { .. } | AttributeSafety::Normal) | None,
-            Safety::Safe(..),
-        ) => {
-            psess.dcx().span_delayed_bug(
-                attr_item.span(),
-                "`check_attribute_safety` does not expect `Safety::Safe` on attributes",
-            );
-        }
-    }
-}
-
 // Called by `check_builtin_meta_item` and code that manually denies
 // `unsafe(...)` in `cfg`
 pub fn deny_builtin_meta_unsafety(diag: DiagCtxtHandle<'_>, unsafety: Safety, name: &Path) {
     // This only supports denying unsafety right now - making builtin attributes
     // support unsafety will requite us to thread the actual `Attribute` through
```

Comment:
is this comment still true?

Reply:
Nope, and it's hasn't been true for a long time :p
Fixed by removing the entire function